grails-app/services/org/pih/warehouse/inventory/OutboundStockMovementService.groovy

```diff
@@ -98,6 +99,20 @@ class OutboundStockMovementService {
                     not {
```

Comment:
Should this `not` be together with the next one or it should be one of these depending on the case if it it approval required or not?

Reply:
Those `not`s are separated by purpose:
For example, it can be read also as
- if approval is required: check if status is in [CREATED, ISSUED, CANCELED]
- if approval is not required: check if the status is in [CREATED, ISSUED, CANCELED, APPROVED, PENDING_APPROVAL, REJECTED]

I just moved the common part higher, and because of that I got rid of one additional else and longer lists of statuses

Reply:
I think it might be easier to follow the code if it was (with some commentary around it):
```
if required {
    not in the list of some statuses + list of other statuses
} else {
    not in the list of some statuses
}
```

Reply:
Are you sure about that? This redundancy of statuses looks odd to me, so maybe it would be enough when I add just comments around the current code?

Reply:
I added some comments for now, if you think it's not enough, let me know

Reply:
Ok, I think this makes sense for now